messages/compiler/cpp/serialize.cpp

```diff
@@ -37,12 +38,16 @@ class BadDataError : public DeserializeError {
 /******************************************************************************
  * Integers
  *
- * All integers are encoded in little endian
+ * All integers are encoded in the byte order specified by cmf::kByteOrder
  ******************************************************************************/
 template <typename T, typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
 void serialize(std::vector<uint8_t>& output, const T& t) {
   if constexpr (std::is_same_v<T, bool>) {
     output.push_back(t ? 1 : 0);
+  } else if constexpr (kByteOrder == ByteOrder::kBigEndian) {
+    for (auto i = sizeof(T); i > 0; i--) {
+      output.push_back(255 & t >> ((i - 1) * 8));
+    }
```

Comment:
nit: add parentheses to clarify precedence

Reply:
Good point, will add, thanks!